captum/attr/_utils/batching.py

```diff
@@ -25,13 +25,16 @@ def _reduce_list(val_list, red_func=torch.cat):
     """
     Applies reduction function to given list. If each element in the list is
     a Tensor, applies reduction function to all elements of the list, and returns
-    the output Tensor / value. If each element is a tuple, applies reduction
+    the output Tensor / value. If each element is a boolean, apply any method (or).
+    If each element is a tuple, applies reduction
     function to corresponding elements of each tuple in the list, and returns
     tuple of reduction function outputs with length matching the length of tuple
     val_list[0].
     """
     if isinstance(val_list[0], torch.Tensor):
         return red_func(val_list)
+    elif isinstance(val_list[0], bool):
+        return any(val_list)
     assert isinstance(val_list[0], tuple), "Elements to be reduced can only be"
     "either Tensors or tuples containing Tensors."
```

Comment:
Yes, _reduce_list expects a list of either tensors or tuples (with all elements corresponding), but we can add support for any iterable if necessary. I think for the line mentioned, it might be fine the way it is, since we intend to reduce all corresponding positioned elements of each tuple (with the assumption that each is the same length), the outer loop just iterates through the positions within a tuple, and the inner call to reduce list obtains all elements of that position.

In your example, the loop should first call _reduce_list with (t,t2,t3) and the second time again with (the second elements) (t, t2, t3). Does that seem correct?

Reply:
Oh, I see, maybe we should document it. Because, I was originally thinking that I can expect the elements only within a tuple to have same dimensionality. Is this used here ?https://github.com/pytorch/captum/blob/0997b63641f8495767f522d5a856697d5c5fe653/captum/attr/_utils/gradient.py#L104

Reply:
Makes sense, will add a comment to make it clearer! I think here each element in gradient_tensors is a tuple matching the input (so each element of the tuple is not necessarily the same dimensionality), but all the corresponding elements of tuples should match.